types/move.ts

```diff
@@ -1,6 +1,10 @@
 import { match } from "ts-pattern";
+import { z } from "zod";
 
-export type Move = "Rock" | "Paper" | "Scissors";
+export const MoveEnum = z.enum(["Rock", "Paper", "Scissors"]);
+export type Move = z.infer<typeof MoveEnum>;
+
+export const move = MoveEnum;
```

Comment:
Normalmente con zod noi chiamiamo lo schema uguale al tipo, con la maiuscola, quindi sarebbe:
```suggestion
export const Move = z.enum(["Rock", "Paper", "Scissors"]);
export type Move = z.infer<typeof Move>;
```

È una convenzione discutibile (visto che di solito gli oggetti non hanno la maiuscola) e non la seguono tutti però da noi si usa quindi direi di fare così.

Motivazioni:
1. tipicamente lo schema e il tipo si corrispondono uno a uno, perché lo schema è l'oggetto che usiamo per validare un input e costruire qualcosa di quel tipo (potrebbero non corrispondere sempre: magari hai più schemi diversi che accettano input diversi ma, dopo il parsing, producono lo stesso tipo – però capita più raramente) -> quindi li chiamiamo uguali per rafforzare questa corrispondenza
2. viene comodo perché quando fai `import { Move } ...` ti importa sia l'oggetto che il tipo

TS distingue i nomi di valori (oggetti, costanti ecc.) e i nomi di tipi, quindi anche se li chiamiamo uguale non c'è ambiguità quando fai riferimento (dal punto di vista di TS – da quello umano bisogna imparare a capire a quale stai facendo riferimento 😅)